server/supernova/server/main.cpp

```diff
@@ -377,8 +377,14 @@ int supernova_main(int argc, char* argv[]) {
                 exit(1);
             }
             EventLoop::run([&server]() { server.run(); });
-        } else
-            server.run_nonrt_synthesis(args);
+        } else {
+            try {
+                server.run_nonrt_synthesis(args);
+            } catch (exception const& e) {
+                cout << "Non-rt synthesis failed: " << e.what() << endl;
+                exit(1);
```

Comment:
I'm not the one to review C++, but how about `exit(EXIT_FAILURE )`? We discussed unifying exit codes in #6728 so hopefully that'll happen at some point.

Reply:
> but how about exit(EXIT_FAILURE)

I chose `exit(1)` for consistency with the remaining code.